web/packages/teleterm/src/ui/services/connectMyComputer/connectMyComputerService.ts

```diff
@@ -15,11 +15,21 @@
  */
 
 import { MainProcessClient } from 'teleterm/mainProcess/types';
+import { TshClient } from 'teleterm/services/tshd/types';
+
+import type * as uri from 'teleterm/ui/uri';
 
 export class ConnectMyComputerService {
-  constructor(private mainProcessClient: MainProcessClient) {}
+  constructor(
+    private mainProcessClient: MainProcessClient,
+    private tshClient: TshClient
+  ) {}
 
   async downloadAgent(): Promise<void> {
     await this.mainProcessClient.downloadAgent();
   }
+
+  createRole(rootClusterUri: uri.RootClusterUri) {
```

Comment:
It's a matter of preference when it comes to annotating return types (I like them because I can easily see what the function returns), but I think we should be consistent. So in this case I would remove `Promise<void>` above or add a type here :)

Btw, explicit return types can also improve the compilation performance https://github.com/microsoft/TypeScript/wiki/Performance#using-type-annotations, but I guess it doesn't really matter in such simple cases.

Reply:
Yeah, I generally prefer annotating return types but I was just lazy here. I'll add the type here.

BTW, I'm not sure if I shared this before, but this is a good article on a similar topic: https://steveklabnik.com/writing/rusts-golden-rule